lib/ops/resources/gravity/gravity.go

```diff
@@ -444,6 +456,19 @@ func (r *Resources) Remove(ctx context.Context, req resources.RemoveRequest) err
 		}
 		r.Printf("TLS key pair %q has been deleted\n", req.Name)
 	case storage.KindSMTPConfig:
+		// SMTP configuration can be deleted only if there is no
+		// alert recipient configured, otherwise it will result
+		// into invalid Alertmanager configuration.
+		alertTargets, err := r.Operator.GetAlertTargets(r.cluster.Key())
+		if err != nil && !trace.IsNotFound(err) {
+			return trace.Wrap(err)
+		}
+		if len(alertTargets) != 0 {
+			return trace.BadParameter("SMTP configuration can " +
```

Comment:
Does this need to make user do something or can we remove the alert target automatically? This might include a warning to the user when deleting the SMTP configuration that the alert target will be also removed?

Reply:
I was considering this but eventually decided against it as it would be an unexpected side-effect (even if there's a warning). In situations like this I prefer to let the user make decision explicitly.